WarshipGame/Assets/Scripts/Gameplay/Ship.cs

```diff
@@ -0,0 +1,117 @@
+    private GlowManager _glowManager;
+    
+    public int MovementPoints => Points;
+    public event Action<Ship> MovementFinished;
+
+    private void Awake()
+    {
+        _glowManager = GetComponent<GlowManager>();
+    }
+
+    internal void Deselect()
+    {
+        _glowManager.ToggleGlow(false);
+    }
+    
+    internal void Select()
+    {
+        _glowManager.ToggleGlow();
+    }
+
+    /// <summary>
+    /// Cycles trough the current path queue and removes the tiles it passed out of the queue
+    /// </summary>
+    /// <param name="currentPath"></param>
+    internal void MoveTroughPath(List<Vector3> currentPath)
+    {
+        _pathPositions = new Queue<Vector3>(currentPath);
+        Vector3 firstTarget = _pathPositions.Dequeue();
+        StartCoroutine(RotationCoroutine(firstTarget, RotationTime, true));
+    }
+
+    /// <summary>
+    /// Rotates from the start location to the end location in a set time
+    /// </summary>
+    /// <param name="endPosition"></param>
+    /// <param name="rotationTime"></param>
+    /// <param name="firstRotation"></param>
+    /// <returns></returns>
+    private IEnumerator RotationCoroutine(Vector3 endPosition, float rotationTime, bool firstRotation = false)
+    {
+        Quaternion startRotation = transform.rotation;
+        endPosition.y = transform.position.y;
+        Vector3 direction = endPosition - transform.position;
+        Quaternion endRotation = Quaternion.LookRotation(direction, Vector3.up);
+
+        if (!Mathf.Approximately(Mathf.Abs(Quaternion.Dot(startRotation, endRotation)),1.0f))
+        {
+            float timeElapsed = 0;
+            while (timeElapsed < rotationTime)
+            {
+                timeElapsed += Time.deltaTime;
+                float lerpStep = timeElapsed / rotationTime;
+                transform.rotation = Quaternion.Lerp(startRotation,endRotation, lerpStep);
+                yield return null;
+            }
+
+            transform.rotation = endRotation;
+        }
+
+        StartCoroutine(MovementCoroutine(endPosition));
+    }
+
+    /// <summary>
+    /// Lerps the player from the start position to the end position in a set Time
+    /// And checks if the Ship reached its end goal.
+    /// </summary>
+    /// <param name="endPosition"></param>
+    /// <returns></returns>
+    private IEnumerator MovementCoroutine(Vector3 endPosition)
+    {
+        Vector3 startPosition = transform.position;
+        endPosition.y = startPosition.y;
+        float timeElapsed = 0;
+        
+        while(timeElapsed < MovementTime)
+        {
+            timeElapsed += Time.deltaTime;
+            float lerpStep = timeElapsed / MovementTime;
+            transform.position = Vector3.Lerp(startPosition,endPosition, lerpStep);
+            yield return null;
+        }
+
+        transform.position = endPosition;
+
+        if (_pathPositions.Count > 0)
+        {   
+            Debug.Log("Selecting the next position");
```

Comment:
`Debug.Log` should not be merged into develop because we are not debugging there